### PR TITLE
cmake/README: Some cleanup

### PR DESCRIPTION
**Pull request by VelocityRa**
## Description

Deletes some duplicated stuff and moves docs for a bunch of CMake vars and some build instructions one directory higher.

## Motivation and Context

They're more visible/accessible this way, they shouldn't have been `cmake/addons` in the first place.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [x] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: cmake/README.md [hunk 1]
```diff
@@ -21,6 +21,38 @@ cmake ... -DENABLE_VAAPI=ON -DENABLE_VDPAU=OFF ...
 
 Unfortunately, Kodi's CMake gazillion options are not fully documented yet. For more information and an updated list of options, please check the main **[CMakeLists.txt](../CMakeLists.txt)**.
 
+## Buildsystem variables
+The buildsystem uses the following variables (which can be passed into it when executing cmake with the -D`<variable-name>=<value>` format) to manipulate the build process (see READMEs in sub-directories for additional variables):
+- `CMAKE_BUILD_TYPE` specifies the type of the build. This can be either *Debug* or *Release* (default is *Release*)
+- `CMAKE_INSTALL_PREFIX` points to the directory where the built add-ons and their additional files (addon.xml, resources, ...) will be installed to (defaults to `<ADDON_DEPENDS_PATH>`)
+- `CMAKE_TOOLCHAIN_FILE` can be used to pass a toolchain file into the add-on builds
+- `CORE_SOURCE_DIR` points to the root directory of the project (default is the absolute representation of ../../.. starting from this directory)
```
Comment (garbear) [inline cmake/README.md]: `../../..` is left over from early cmake days, when CMakeLists.txt was located at project/cmake/CMakeLists.txt. Now the path would be `..`, and there's probably a better way of wording this.

Comment (VelocityRa) [reply cmake/README.md]: Reworded.

File: cmake/README.md [hunk 2]
```diff
@@ -21,6 +21,38 @@ cmake ... -DENABLE_VAAPI=ON -DENABLE_VDPAU=OFF ...
 
 Unfortunately, Kodi's CMake gazillion options are not fully documented yet. For more information and an updated list of options, please check the main **[CMakeLists.txt](../CMakeLists.txt)**.
 
+## Buildsystem variables
+The buildsystem uses the following variables (which can be passed into it when executing cmake with the -D`<variable-name>=<value>` format) to manipulate the build process (see READMEs in sub-directories for additional variables):
+- `CMAKE_BUILD_TYPE` specifies the type of the build. This can be either *Debug* or *Release* (default is *Release*)
+- `CMAKE_INSTALL_PREFIX` points to the directory where the built add-ons and their additional files (addon.xml, resources, ...) will be installed to (defaults to `<ADDON_DEPENDS_PATH>`)
+- `CMAKE_TOOLCHAIN_FILE` can be used to pass a toolchain file into the add-on builds
+- `CORE_SOURCE_DIR` points to the root directory of the project (default is the directory above the current one)
```
Comment (Rechi) [inline cmake/README.md]: only useful to set for addons

File: cmake/README.md [hunk 3]
```diff
@@ -21,6 +21,38 @@ cmake ... -DENABLE_VAAPI=ON -DENABLE_VDPAU=OFF ...
...
+- `BUILD_DIR` points to the directory where the add-ons and their dependencies will be downloaded and built
+- `ARCH_DEFINES` specifies the platform-specific C/C++ preprocessor defines (defaults to empty)
+
+## Deprecated buildsystem variables
+Buildsystem will print a warning if you use any of the below-listed variables. For now they still work but you should adapt your workflow to the new variables.
+- `APP_ROOT` - Use `CORE_SOURCE_DIR` instead
```
Comment (Rechi) [inline cmake/README.md]: was only used for addons

File: cmake/README.md [hunk 4]
```diff
@@ -21,6 +21,38 @@ cmake ... -DENABLE_VAAPI=ON -DENABLE_VDPAU=OFF ...
...
+
+CMake supports multiple generators. See [here] (https://cmake.org/cmake/help/v3.1/manual/cmake-generators.7.html) for a list.
+
+In case of additional options the call might look like this:
+
+cmake `<path>` [-G `<generator>`] \  
+      -DCMAKE_BUILD_TYPE=Release \  
+      -DCORE_SOURCE_DIR="`<path-to-app-root>`" \  
+      -DARCH_DEFINES="-DTARGET_LINUX" \  
+      -DADDON_DEPENDS_PATH=`<path-to-built-depends>` \
```
Comment (Rechi) [inline cmake/README.md]: only used for addons

File: cmake/addons/README.md [hunk 5]
```diff
@@ -24,42 +24,13 @@ Available platforms are: linux, windows, osx, ios, android, rbpi and freebsd.
...
      - a regular expression that every `<addon-id>` is matched against (e.g. `ADDONS_TO_BUILD="pvr.*"`) to build all pvr add-ons
 - `ADDONS_DEFINITION_DIR` points to the directory containing the definitions for the addons to be built
 - `ADDON_SRC_PREFIX` can be used to override the add-on repository location. It must point to the locally available parent directory of the add-on(s) to build. `<addon-id>` will be appended to this path automatically
-- `CMAKE_BUILD_TYPE` specifies the type of the build. This can be either *Debug* or *Release* (default is *Release*)
-- `CMAKE_INSTALL_PREFIX` points to the directory where the built add-ons and their additional files (addon.xml, resources, ...) will be installed to (defaults to `<ADDON_DEPENDS_PATH>`)
-- `CMAKE_TOOLCHAIN_FILE` can be used to pass a toolchain file into the add-on builds
 - `ADDON_DEPENDS_PATH` points to the directory containing the *include* and *lib* directories of the add-ons' dependencies.
-- `CORE_SOURCE_DIR` points to the root directory of the project (default is the absolute representation of ../../.. starting from this directory)
```
Comment (Rechi) [inline cmake/addons/README.md]: only useful to set for addons

File: cmake/addons/README.md [hunk 6]
```diff
@@ -24,42 +24,13 @@ Available platforms are: linux, windows, osx, ios, android, rbpi and freebsd.
...
-- `BUILD_DIR` points to the directory where the add-ons and their dependencies will be downloaded and built
 - `PACKAGE_ZIP=ON` means that the add-ons will be 'packaged' into a common folder, rather than being placed in `<CMAKE_INSTALL_PREFIX>/lib/kodi/addons` and `<CMAKE_INSTALL_PREFIX>/share/kodi/addons`
 - `PACKAGE_DIR` points to the directory where the ZIP archived add-ons will be stored after they have been packaged (defaults to `<BUILD_DIR>/zips`)
-- `ARCH_DEFINES` specifies the platform-specific C/C++ preprocessor defines (defaults to empty)
 - `ADDON_TARBALL_CACHING` specifies whether downloaded add-on source tarballs should be cached or not (defaults to *ON*)
-
-## Deprecated buildsystem variables
-Buildsystem will print a warning if you use any of the below-listed variables. For now they still work but you should adapt your workflow to the new variables.
-- `APP_ROOT` - Use `CORE_SOURCE_DIR` instead
```
Comment (Rechi) [inline cmake/addons/README.md]: was only used for addons

**Review by Rechi — COMMENTED** [review]
I don't think it is useful to move vars from addon to core readme if they aren't used in core.

**Comment by VelocityRa** [comment]
Ok, should be good now.

File: cmake/README.md [hunk 7]
```diff
@@ -21,6 +21,32 @@ cmake ... -DENABLE_VAAPI=ON -DENABLE_VDPAU=OFF ...
 
 Unfortunately, Kodi's CMake gazillion options are not fully documented yet. For more information and an updated list of options, please check the main **[CMakeLists.txt](../CMakeLists.txt)**.
 
+## Buildsystem variables
+The buildsystem uses the following variables (which can be passed into it when executing cmake with the -D`<variable-name>=<value>` format) to manipulate the build process (see READMEs in sub-directories for additional variables):
+- `CMAKE_BUILD_TYPE` specifies the type of the build. This can be either *Debug* or *Release* (default is *Release*)
+- `CMAKE_INSTALL_PREFIX` points to the directory where the built add-ons and their additional files (addon.xml, resources, ...) will be installed to (defaults to `<ADDON_DEPENDS_PATH>`)
```
Comment (Rechi) [inline cmake/README.md]: explanation isn't true for Kodi

File: cmake/README.md [hunk 8]
```diff
@@ -21,6 +21,32 @@ cmake ... -DENABLE_VAAPI=ON -DENABLE_VDPAU=OFF ...
...
+- `CMAKE_TOOLCHAIN_FILE` can be used to pass a toolchain file into the add-on builds
```
Comment (Rechi) [inline cmake/README.md]: this isn't restricted to add-on builds

File: cmake/README.md [hunk 9]
```diff
@@ -21,6 +21,32 @@ cmake ... -DENABLE_VAAPI=ON -DENABLE_VDPAU=OFF ...
...
+- `BUILD_DIR` points to the directory where the add-ons and their dependencies will be downloaded and built
```
Comment (Rechi) [inline cmake/README.md]: only used by addons

File: cmake/README.md [hunk 10]
```diff
@@ -21,6 +21,32 @@ cmake ... -DENABLE_VAAPI=ON -DENABLE_VDPAU=OFF ...
...
+- `ARCH_DEFINES` specifies the platform-specific C/C++ preprocessor defines (defaults to empty)
+
+## Building
+The buildsystem makes some assumptions about the environment which must be met by whoever uses it:
+- Any dependencies of the add-ons must already be built and their include and library files must be present in the path pointed to by `<CMAKE_PREFIX_PATH>` (in *include* and *lib* sub-directories)
```
Comment (Rechi) [inline cmake/README.md]: only mentions requirements for add-ons

File: cmake/README.md [hunk 11]
```diff
@@ -21,6 +21,32 @@ cmake ... -DENABLE_VAAPI=ON -DENABLE_VDPAU=OFF ...
...
+
+`cmake <path> -G <generator>`
+
+CMake supports multiple generators. See [here] (https://cmake.org/cmake/help/v3.1/manual/cmake-generators.7.html) for a list.
+
+In case of additional options the call might look like this:
+
+cmake `<path>` [-G `<generator>`] \  
+      -DCMAKE_BUILD_TYPE=Release \  
+      -DCORE_SOURCE_DIR="`<path-to-app-root>`" \
```
Comment (Rechi) [inline cmake/README.md]: never set this for Kodi

**Review by Rechi — COMMENTED** [review]
missed a few things in my previous review

File: cmake/README.md (hunk 7 above)
Comment (VelocityRa) [reply cmake/README.md]: What should it say then?

Comment (Rechi) [reply cmake/README.md]: add the info what it does for Kodi

Comment (VelocityRa) [reply cmake/README.md]: I just moved it back to the addon dir.

**Comment by MartijnKaijser** [comment]
Fail unrelated

**Pull request merged**